packages/cli/src/rpc/explain/explain.ts

```diff
@@ -72,28 +72,28 @@ export class Explain extends EventEmitter {
     await navie.ask(this.question, this.codeSelection);
   }
 
-  async searchContext(data: SearchContextOptions): Promise<SearchContextResponse> {
-    const { vectorTerms } = data;
+  async searchContext(data: ContextV2.ContextRequest): Promise<ContextV2.ContextResponse> {
+    let { keywords } = data;
     let { tokenCount } = data;
+
     if (!tokenCount) {
-      warn(chalk.bold(`Warning: tokenCount not set, defaulting to ${DEFAULT_TOKEN_LIMIT}`));
+      warn(chalk.bold(`Warning: Token limit not set, defaulting to ${DEFAULT_TOKEN_LIMIT}`));
       tokenCount = DEFAULT_TOKEN_LIMIT;
     }
+    if (!keywords || keywords.length === 0) {
+      warn(chalk.bold(`Warning: No keywords provided, context result may be unpredictable`));
+      keywords = [];
+    }
 
-    this.status.vectorTerms = vectorTerms;
+    this.status.vectorTerms = keywords;
 
     this.status.step = ExplainRpc.Step.SEARCH_APPMAPS;
 
     // TODO: More accurate char limit? Probably doesn't matter because they will be
     // pruned by the client AI anyway.
     // The meaning of tokenCount is "try and get at least this many tokens"
     const charLimit = tokenCount * 3;
-    const searchResult = await collectContext(
-      this.directories,
-      this.appmaps,
-      vectorTerms,
-      charLimit
-    );
+    const searchResult = await collectContext(this.directories, this.appmaps, keywords, charLimit);
```

Comment:
This can be refactored to return an array instead of an object, because we are converting it into an array anyway.